### PR TITLE
Part 1: Backend parkin parsing

### PR DESCRIPTION
**Pull request by lancetipton**
> This is part 1 of keg-herkin updates
> All these PRs share the same tests
> But the code has been broken up into smaller PRs for easier review

## Context

* Keg-Herkin backend API needs some cleanup work

## Goal

* Clean up the backend API
* Added a results synced folder

## Updates

* Cleaned up the dockerfile
  * Remove expo, it's no longer being used
* Added a nodemon config file for ignore specific folders
* Updated all the deps in the root `package.json` file
 * Cleaned up a number of endpoints to be more consistent
* Removed old parser code and replaced with `parkin`
  * Updated feature and definitions to be parsed with `parkin `
* Added `testType` property to the fileTree nodes
  * Allows us to know what type of file the node relates to
  * Helps with knowing what  editor to load and tests to run from the front end
  * Added `resolveTestFileType` helper to parse the node path to know the `testType`
* Converted all the testUtils step definitions to use `cjs`
  * This allows them to be loaded in node when parsing `step-definitions `

## Testing

* Do the tests in [this PR](https://github.com/simpleviewinc/keg-herkin/pull/32)

### CODE REVIEW

File: repos/backend/libs/definitions/definitionsParser.js [hunk 1]
```diff
@@ -1,106 +1,84 @@
...
+      // We still want to load the file content
+      // Even if the require call fails
+      // So wrap if in a try catch, and log the error if it happends
+      let response
+      try {
+
+        // Always clear out the node require cache
+        // This ensure we get a fresh file every time
+        // Otherwise changed files would not get reloaded
+        if(require.cache[filePath])
```
Comment (mpcarolin) [inline repos/backend/libs/definitions/definitionsParser.js:47]: pretty sure checking this condition is unnecessary. According to [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete):

> If the property which you are trying to delete does not exist, delete will not have any effect and will return true.

File: repos/backend/libs/definitions/definitionsParser.js [hunk 2]
```diff
@@ -1,106 +1,84 @@
...
+          delete require.cache[filePath]
+
+        // Require the file, to auto-load the definitions into parkin
+        // Later we'll pull them from parkin
+        response = require(filePath)
```
Comment (mpcarolin) [inline repos/backend/libs/definitions/definitionsParser.js:52]: Might be a good case for a jsutils function at some point:
```js
const cleanRequire = filePath => {
  delete require.cache[filePath]
  return require(filePath)
}
```

**Review by mpcarolin — APPROVED** [review]
Looks good, but I left one suggestion

File: repos/backend/libs/definitions/definitionsParser.js (hunk 1 above)
Comment (lancetipton) [reply repos/backend/libs/definitions/definitionsParser.js:47]: Hmm... nice, didn't know that actually. Although I not sure what's faster...
Check if the property exists, or trying to delete something that doesn't exist

File: repos/backend/libs/definitions/definitionsParser.js (hunk 2 above)
Comment (lancetipton) [reply repos/backend/libs/definitions/definitionsParser.js:52]: Yah that would be a good one. Although, it's not often I need to do something like this. The next time I need it, I'll probably forget the method exists in jsutils...lol

**Pull request merged**